test/scripts/boot-image

```diff
@@ -90,6 +95,106 @@ def ensure_uncompressed(filepath):
         yield filepath
 
 
+@contextlib.contextmanager
+def ensure_qcow2(filepath):
+    base, ext = os.path.splitext(filepath)
+    if ext != ".qcow2":
+        print(f"Converting {filepath} to qcow2")
+        filepath_qcow2 = base + ".qcow2"
+        testlib.runcmd_nc([
+            "qemu-img", "convert",
+            "-f", "raw",
+            "-O", "qcow2",
+            "-o", "preallocation=metadata",
+            filepath, filepath_qcow2,
+        ])
+        yield filepath_qcow2
+        os.unlink(filepath_qcow2)
+    else:
+        yield filepath
+
+
+@contextlib.contextmanager
+def make_cloud_init_iso(ssh_key) -> pathlib.Path:
+    with TemporaryDirectory() as tmpdir:
+        user_data = pathlib.Path(tmpdir) / "user-data.yaml"
+        user_data_content = textwrap.dedent(f"""\
+        #cloud-config
+        users:
+          - name: root
+            ssh_authorized_keys:
+              - {ssh_key}
+        """)
+        user_data.write_text(user_data_content)
+        meta_data = pathlib.Path(tmpdir) / "meta-data"
+        meta_data.write_text('{"instance-id": "i-1234567890abcdef0"}')
+        iso_path = pathlib.Path(tmpdir) / "cloud-init.iso"
+        subprocess.check_call(
+            ["cloud-localds", os.fspath(iso_path), user_data.name, meta_data.name],
+            cwd=tmpdir,
+        )
+        yield iso_path
+
+
+def can_run_qemu_test(arch, image_path, config_file):
```

Comment:
Do we also need/want to test if `cloud-init` is actually in the image, since it's used to write keys and such?